demisto_sdk/commands/download/downloader.py

```diff
@@ -439,305 +460,397 @@ def build_req_params(self):
+        for playbook in content_items:
             endpoint = f"/playbook/{playbook}/yaml"
             try:
-                api_response = demisto_client.generic_request_func(
-                    self.client, endpoint, req_type, response_type="object"
-                )
+                api_response: dict = demisto_client.generic_request_func(
+                    self.client, endpoint, "GET", response_type="object",
+                )[0]
+
             except ApiException as err:
                 # handling in case the id and name are not the same,
                 # trying to get the id by the name through a different api call
-                if playbook_id := self.get_playbook_id_by_playbook_name(playbook):
-                    endpoint = f"/playbook/{playbook_id}/yaml"
-                    api_response = demisto_client.generic_request_func(
-                        self.client, endpoint, req_type, response_type="object"
-                    )
-                else:
-                    raise err
-            playbook_list.append(yaml.load(api_response[0].decode()))
+                logger.debug(f"API call using playbook's name failed:\n{err}\n"
+                             f"Attempting to fetch using playbook's ID...")
+
+                playbook_id = self.get_playbook_id_by_playbook_name(playbook)
 
-        return playbook_list
+                if not playbook_id:
+                    raise
 
-    def arrange_response(self, system_items_list):
-        if self.system_item_type in ["Classifier", "Mapper"]:
-            system_items_list = system_items_list.get("classifiers")
+                logger.debug(f"Found matching ID for '{playbook}' - {playbook_id}.\n"
+                             f"Attempting to fetch playbook's YAML file using the ID.")
 
-        return system_items_list
+                endpoint = f"/playbook/{playbook_id}/yaml"
+                api_response = demisto_client.generic_request_func(
+                    self.client, endpoint, "GET",  _preload_content=False,
+                )[0]
+
+            if not isinstance(api_response, dict):
+                # If there are API issues, it might return HTML, which is returned as a string.
+                raise ApiException(f"Unexpected response from server:\n"
+                                   f"'{api_response}'.")
+
+            downloaded_playbooks.append(api_response)
 
-    def build_file_name(self, item):
-        item_name: str = item.get("name") or item.get("id")
+        logger.debug(f"'{len(downloaded_playbooks)}' system playbooks were downloaded successfully.")
+        return downloaded_playbooks
+
+    def build_file_name(self, content_item: dict, content_item_type: str) -> str:
+        item_name: str = content_item.get("name") or content_item.get("id")
         return (
-            item_name.replace("/", " ").replace(" ", "_")
-            + ITEM_TYPE_TO_PREFIX[self.system_item_type]
+            item_name.replace("/", "_").replace(" ", "_")
```

Comment:
It's a pre-existing function to create content file names for content items from their content name.
Although I do notice now that if a content item is named `Michael test / test`, it will be return `Michael_test___test`

I'll update that (to return `Michae_test_test` in that case, for example)